tests/components/withings/__init__.py

```diff
@@ -43,19 +48,23 @@ async def call_webhook(
 class MockWithings:
     """Mock object for Withings."""
 
-    def __init__(self, user_profile: ProfileConfig):
+    def __init__(
+        self,
+        device_fixture: str = "person0_get_device.json",
+        measurement_fixture: str = "person0_get_meas.json",
+        sleep_fixture: str = "person0_get_sleep.json",
+        notify_list_fixture: str = "person0_notify_list.json",
+    ):
         """Initialize mock."""
-        self.api_response_user_get_device = user_profile.api_response_user_get_device
-        self.api_response_measure_get_meas = user_profile.api_response_measure_get_meas
-        self.api_response_sleep_get_summary = (
-            user_profile.api_response_sleep_get_summary
-        )
+        self.device_fixture = device_fixture
+        self.measurement_fixture = measurement_fixture
+        self.sleep_fixture = sleep_fixture
+        self.notify_list_fixture = notify_list_fixture
 
     def user_get_device(self) -> UserGetDeviceResponse:
         """Get devices."""
-        if isinstance(self.api_response_user_get_device, Exception):
-            raise self.api_response_user_get_device
-        return self.api_response_user_get_device
+        fixture = json.loads(load_fixture(f"withings/{self.device_fixture}"))
```

Comment:
You can use `load_json_object_fixture` (and `load_json_array_fixture` for notify_list)